stl/inc/thread

```diff
@@ -215,7 +216,11 @@ private:
     friend thread::id thread::get_id() const noexcept;
     friend thread::id this_thread::get_id() noexcept;
     friend bool operator==(thread::id _Left, thread::id _Right) noexcept;
+#if _HAS_CXX20
+    friend strong_ordering operator<=>(thread::id _Left, thread::id _Right) noexcept;
+#else // ^^^ _HAS_CXX20 / !_HAS_CXX20 vvv
     friend bool operator<(thread::id _Left, thread::id _Right) noexcept;
+#endif // !_HAS_CXX20

```

Comment:
Would it make sense to default the `operator==` for C++20 and make them hidden friends?

Reply:
They can't be hidden friends; [the difference is observable](https://godbolt.org/z/TxPq44). Arguably `operator==` also cannot be defaulted because doing so would make it `constexpr` and we're forbidden to make things `constexpr` where the Standard doesn't depict them as such.